var/spack/repos/builtin/packages/py-opentuner/package.py

```diff
@@ -0,0 +1,43 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class PyOpentuner(PythonPackage):
+    """An extensible framework for program autotuning."""
+
+    homepage = "http://opentuner.org/"
+    url      = "https://github.com/jansel/opentuner"
+
+    version('0.8.0', git='https://github.com/jansel/opentuner', commit='4cb9135')
+
+    # No support for Python 3 yet
+    depends_on('python@2.7:2.8')
+
+    depends_on('py-sqlalchemy@0.8.2:')
+    depends_on('py-numpy@1.8.0:')
+    depends_on('py-argparse@1.2.1:')
+    depends_on('py-pysqlite@2.6.3:')
+    depends_on('py-fn@0.2.12:')
```

Comment:
There are optional dependencies (`optional-requirements.txt`)
```
django==1.6.1
matplotlib==1.1.1
virtualenv==1.9.1
```
not sure if it's worth adding these as variants though.

Reply:
I think there is no django package in spack, and adding matplotlib alone doesn't make sense as the virtualization tool requires both. virtualenv isn't really necessary in spack, right?